dallinger/experiment.py

```diff
@@ -723,14 +732,15 @@ def on_assignment_submitted_to_recruiter(self, participant, event):
 
         # If they pass the data check, we might pay a bonus
         bonus = self.bonus(participant=participant)
-        participant.bonus = bonus
-        if bonus >= min_real_bonus:
+        has_already_received_bonus = participant.bonus is not None
+        if not has_already_received_bonus and bonus >= min_real_bonus:
             self.log("Bonus = {}: paying bonus".format(bonus))
             participant.recruiter.reward_bonus(
                 participant,
                 bonus,
                 self.bonus_reason(),
             )
+            participant.bonus = bonus
         else:
```

Comment:
Better to disambiguate the two reasons why a bonus might not be paid in the log message (i.e. bonus < min_real_bonus vs has_already_received_bonus)